### ISSUE TITLE
Login does not work on RPi based Homebridge setup

### ISSUE THREAD

**Issue by RienduPre** [mkellsy/homebridge-config-ui#21]
Hi,

I cannot logon using the webinterface. The logon screen freezes for a while and errors out.

It works on a OSX based Homebridge, but it does noet work on any RPi based Homebridge

The log says:
[9/20/2017, 8:07:33 PM] [Config] Console is listening on port 8080.
[9/20/2017, 8:07:33 PM] Homebridge is running on port 51826.
[9/20/2017, 8:07:57 PM] [Config] Administrator successfully logged in.

No other errors however.

Any idea why this behavior does only occur on an RPi?

Thanks,
Rien

**Comment by chriskmn** [comment]
Same here :(

**Comment by RienduPre** [comment]
Replace Line 186 to 189 in npm.js and the problem is solved.


             ```
   if (fs.existsSync("/usr/local/lib/node_modules")) {
                    base = "/usr/local/lib/node_modules";
                } else {
                    base = "/usr/lib/node_modules";

```

**Comment by chriskmn** [comment]
great, this works for me !!! Tahnk you  @RienduPre 

Can anybody update that to git ?

**Comment by shidevil** [comment]
@RienduPre good job on fixing it as there are too many configuration. i will update my pull request.

**Comment by shidevil** [comment]
Please test if its working :)

**Comment by Bankaifan** [comment]
Testet ... it works!

**Comment by shidevil** [comment]
@Bankaifan thats great. I will ask @mkellsy to push it.

**Comment by itsmepetrov** [comment]
@shidevil please, add an ability to specify `base` path in config, because in my case the `base` path is `/opt/lib/node_modules` (I am running it on open-wrt router)

**Comment by itsmepetrov** [comment]
@shidevil and make it line [config.js#L121](https://github.com/shidevil/homebridge-config-ui/blob/master/routes/config.js#L121) safer, because some config files may not contain `accessories ` field (like mine)

**Comment by shidevil** [comment]
@itsmepetrov I will look into it. It is definitely possible.  I will also check on Line 121. Yes i do agree that some may not contain accessories field.

**Comment by shidevil** [comment]
@itsmepetrov updated config.js and also specify base path in config file. Please check.

**Comment by smulle48** [comment]
@RienduPre can you point out exactly which lines that need to be replaced? don't know if I got the same npm file, so I just want to make sure I replace the correct lines :-)

**Comment by mkellsy** [comment]
I will commit this, however I need to seperate Fix #6, #15, #18. The temperature code, needs to be reviewed.

**Comment by mkellsy** [comment]
In the platform section for config, modify this value. "base": "/usr/local/lib/node_modules". Set the base to where your global node_modules are. You may have to Google your operating system.
